src/plugins/embeddable/public/lib/actions/apply_filter_action.test.ts

```diff
@@ -93,7 +93,7 @@ const getEmbeddable = () => {
 
 describe('execute()', () => {
   describe('when no filters are given', () => {
-    test('throws an error', async () => {
+    test.skip('throws an error', async () => {
```

Comment:
@streamich I'm not sure if you're the right person to ping about this (blame `git blame` 😅), but I had to skip these tests as they were producing unhandled promise rejections due to an uncaught exception.

Reply:
Here's the stack trace of the unhandled promise rejection:

```
Error: Applying a filter requires a filter and embeddable as context
    at Object.execute (./src/plugins/embeddable/public/lib/actions/apply_filter_action.ts:58:15)
    at error (./src/plugins/embeddable/public/lib/actions/apply_filter_action.test.ts:104:16)
    at expectError (./src/plugins/embeddable/public/tests/helpers.ts:28:5)
    at Object.test.only (./src/plugins/embeddable/public/lib/actions/apply_filter_action.test.ts:103:21)
    at Object.asyncJestTest (./node_modules/jest-jasmine2/build/jasmineAsyncInstall.js:102:37)
    at resolve (./node_modules/jest-jasmine2/build/queueRunner.js:43:12)
    at new Promise (<anonymous>)
    at mapper (./node_modules/jest-jasmine2/build/queueRunner.js:26:19)
    at promise.then (./node_modules/jest-jasmine2/build/queueRunner.js:73:41)
```

I don't see a reason to try and fix this before this PR is merged as the test fails already on the 2nd line, but the failure is just suppressed due to the unhandled rejection not being taken seriously.

Reply:
Yea, this is pretty funny, the `expectError` function is wrong - at least it always returns an Error object no matter what.  I have a fix for this test and the function here: https://github.com/elastic/kibana/pull/52078 . Since I changed the fn it may end up having to be a bigger PR... waiting to see what ci tells me.  cc @streamich 

My concern with skipping is that all these tests will be forgotten about, like the Embeddables which has been turned off for four months in order to get a PR merged (I can't even find the issue now...). I do see your point about the tests being useless now though.  We really just need a better way to solve the issue with skipped tests being forgotten about and not prioritized.

Reply:
This test is "useless" in the sense it doesn't check anything because it's currently aborted before it even starts to assert anything - and so skipping it doesn't reduce test coverage.

No one is arguing that we should leave tests skipped. Skipped tests should normally be avoided like the plague. But the pros of merging this PR with this test skipped far outweigh the cons as we currently can't trust any of our tests with `async` functions. This, in turn, might introduce even bigger and more serious errors as a result. And a skipped test is at least more visible than a green test that silently fails.

As described in the PR description, I'll make issues and assign them to the appropriate teams to get the test un-skipped once this PR lands. But if you feel like fixing this bug already now, please push your fixes to this PR to avoid merge conflicts and so this PR isn't blocked.

Reply:
> This test is "useless" in the sense it doesn't check anything because it's currently aborted before it even starts to assert anything - and so skipping it doesn't reduce test coverage.

Yes, I understand, it's useless in its *current form*, didn't mean to imply otherwise. 

Perhaps I was unclear, my comment about skipped tests was just a comment about concern with them in general, not a decision on the PR (and actually, it seems like the team has been doing well fixing `skipped-test` issues - only a couple outstanding, so maybe my concern was wrong).  I am fine if you merge this PR with these kinds of tests skipped - ones where skipping doesn't actually reduce test coverage. I only looked at this one test though - @streamich mentioned some other skipped ones did still provide test coverage.  It was my understanding you and he synced after the PR description was written and the agreed upon outcome was that the app arch team would fix the tests first, before the PR was merged.  It sounds like that perhaps there wasn't actually a clear understanding of that.

As far as that other PR goes - I don't think it should go in this one because it might grow in scope if I hit PR failures (I suspect it will since I fixed the `expectError` fn).  By keeping it separate, you can merge before I finish that PR. Keeping them separate actually helps not block your PR, imo.  Also, it shouldn't cause merge conflicts - we haven't edited any of the same lines. If I merge that PR first, you'd just have to undo the skip. If you merge first, I'll undo the skip.  Thoughts?

Reply:
As it turns out, that fix was easier than I thought, required no other changes.  I'd still like to go ahead and merge that separately.  Then that test fix won't be held up by this PR.  Like I said though, it really shouldn't cause any merge conflicts since all you touched in here was adding the `skip`.  If it does cause an issue though, lmk, I'll help resolve.

Reply:
Good to hear it was an easy fix. I'll remove the `skip` and rebase with `master` to get your fix in.

Reply:
Rebased 👍